seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-data-structure-questions.json

```diff
@@ -45,23 +45,25 @@
       "id": "587d8250367417b2b2512c5e",
       "title": "Learn how a Stack Works",
       "description": [
-        "You are probably familiar with stack of books on your table. You might have used the undo feature in your computer. You are also used to hitting the back button on your phone to go back to the previous view in your app.",
+        "You are probably familiar with stack of books on your table. You might have used the undo feature in your computer. You are also probably used to hitting the back button on your phone to go back to the previous view in your app.",
         "You know what they all have in common? They all store the data in a way so that you can traverse backwards.",
-        "The topmost book in the stack was the one that was put there at the last. If you remove that book from your stack's top, you would expose the book that was put there before the last book and so on.",
-        "If you think about it, in all the above examples, you are getting Last-In-First-Out type of service. We will try to mimic this with our code - create a similar data storage scheme with JS arrays and functions that we always get back first what we put there last.",
-        "Oh, and did I mention such data storage scheme is called Stack! In particular, we would have to implement the push function that pushes JS object at the top of the stack; and pop function, that removes the JS object that's at the top of the stack at the current moment.",
-        "Instructions",
-        "Here we have a stack of homework assignments represented as an array: BIO12 is at the base, and PSY44 is at the top of the stack.",
-        "Remove the element on top of the stack, and Add CS50."
+        "The topmost book in the stack was the one that was put there last. If you remove that book from your stack's top, you would expose the book that was put there before the last book and so on.",
+        "If you think about it, in all the above examples, you are getting <dfn>Last-In-First-Out</dfn> type of service. We will try to mimic this with our code.",
+        "Oh, and did I mention such a data storage scheme is called a <dfn>Stack</dfn>! In particular, we would have to implement the <code>push()</code> method that pushes JavaScript objects at the top of the stack; and <code>pop()</code> method, that removes the JavaScript object that's at the top of the stack at the current moment.",
+        "<hr>",
+        "Here we have a stack of homework assignments represented as an array: <code>\"BIO12\"</code> is at the base, and <code>\"PSY44\"</code> is at the top of the stack.",
+        "Create a stack data storage scheme with JavaScript arrays and functions that we always get back first what we put there last.",
```

Comment:
This challenge only requires you to use `pop` and `push`. You don't need a function to pass the challenge. The tests only look at `homeworkStack`. I can change it something like

> Modify the given array and treat it like a `stack` using JavaScript methods. Remove the top element `"BIO12"` from the stack. Then add `"CS50"` to be the new top element of the stack. 

But now that I look at it, the unit tests need a bit of work. You can pass the challenge if you just set `homeworkStack` to the correct array elements.